applications/GeoMechanicsApplication/custom_utilities/hydraulic_discharge.cpp

```diff
@@ -0,0 +1,50 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf
+//
+
+#include "custom_utilities/hydraulic_discharge.h"
+
+namespace Kratos
+{
+void HydraulicDischarge::CalculateHydraulicDischarge(const std::vector<array_1d<double, 3>>& rFluidFlux,
+                                                     const std::vector<double>& rIntegrationCoefficients,
+                                                     const Geometry<Node>::ShapeFunctionsGradientsType& rdNDxContainer,
+                                                     GeometryData::IntegrationMethod IntegrationMethod,
+                                                     Geometry<Node>& rGeometry)
+{
+    KRATOS_TRY
+
+    const IndexType number_of_integration_points = rGeometry.IntegrationPointsNumber(IntegrationMethod);
+    const auto dimension        = rGeometry.WorkingSpaceDimension();
+    const auto number_of_points = rGeometry.PointsNumber();
+    Matrix     grad_Np_T(number_of_points, dimension);
+
+    for (unsigned int integration_point = 0; integration_point < number_of_integration_points; ++integration_point) {
+        noalias(grad_Np_T) = rdNDxContainer[integration_point];
+
+        auto integration_coefficient = rIntegrationCoefficients[integration_point];
```

Comment:
Missed this earlier, bit nitpicky, but this one can be const:
```suggestion
        const auto integration_coefficient = rIntegrationCoefficients[integration_point];
```

Reply:
done